### PR TITLE
Add Scikit Learn Choosing the Right Estimators cheatsheet

### PR DESCRIPTION
**Pull request by hstsethi**
## What does this PR do?
Add cheatsheet

### Description

SVG cheatsheet from scikit-learn doc. The reason it's in a separate subsection because the list is focused on machine learning. There is nothing much related to vanilla Python.

### Why is this valuable (or not)?

Yes it is. Choosing the right estimator is a really difficult task. Multiple one have to be tested, which is time consuming.

### How do we know it's really free?

It's from official scikit learn documentation which is an open source library.

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction).
- [X] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
you should add https://scikit-learn.org/stable/user_guide.html as book, not chapter 12 as a cheatsheet. it should be in machine learning

**Comment by hstsethi** [comment]
Thank you for your feedback. I have done as you said so.

But now I'm getting this false positive linter error. The title of web page is just "User Guide" which is ambiguous. So I appended "[Scikit Learn]" before the title. 

Technically, `sc` should come before `sp`. But due to introduction of "[]", maybe linter got it wrong. 

Can you please tell me how to proceed? 
@eshellman

**Comment by eshellman** [comment]
You have discovered that our linter wants to include punctuation in its sorting - that's just what it does.

One other source for titles is the page's title tag, which in this case contains "User Guide — scikit-learn 1.5.2 documentation".
So I would use "User Guide — scikit-learn" because we're not interested in tracking the version number. One reason we use titles from the resource is that we want contributors to pick the same title years from now and avoid duplications.

**Comment by hstsethi** [comment]
> You have discovered that our linter wants to include punctuation in its sorting - that's just what it does.
> 
> One other source for titles is the page's title tag, which in this case contains "User Guide — scikit-learn 1.5.2 documentation". So I would use "User Guide — scikit-learn" because we're not interested in tracking the version number. One reason we use titles from the resource is that we want contributors to pick the same title years from now and avoid duplications.

Done.
Please review again.

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -452,6 +452,7 @@ Books that cover a specific programming language can be found in the [BY PROGRAM
 * [The Python Game Book](https://web.archive.org/web/20210308080726/https://thepythongamebook.com/en%3Astart) - Horst Jens *(:card_file_box: archived)*
 * [Top 10 Machine Learning Algorithms Every Engineer Should Know](https://www.dezyre.com/article/top-10-machine-learning-algorithms/202) - Binny Mathews, Omair Aasim
 * [Understanding Machine Learning: From Theory to Algorithms](https://www.cs.huji.ac.il/~shais/UnderstandingMachineLearning) - Shai Shalev-Shwartz, Shai Ben-David
+* [User Guide - scikit-learn](https://scikit-learn.org/stable/user_guide.html) - Scikit-learn developers. (HTML) (BSD)
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:455]:
```suggestion
* [User Guide - scikit-learn](https://scikit-learn.org/stable/user_guide.html) - Scikit-learn developers (HTML) (BSD)
```
one final nit

**Pull request merged**